services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/X509CertificateCredential.java

```diff
@@ -198,4 +258,45 @@ public final X509CertificateCredential setSecrets(final List<X509CertificateSecr
         this.secrets.addAll(secrets);
         return this;
     }
+
+    /**
+     * Sets the <em>generated-auth-id</em> by applying the <em>auth-id-template</em> from the tenant's
+     * trust anchor to the client certificate's subject DN.
+     * <p>
+     * It is only applicable, if a template is configured.
+     *
+     * @param tenant The tenant information.
+     * @throws NullPointerException if the tenant is {@code null}.
+     */
+    @JsonIgnore
+    public final void applyAuthIdTemplate(final Tenant tenant) {
+        Objects.requireNonNull(tenant, "tenant information must not be null");
+
+        Optional.ofNullable(issuerDN)
+                .flatMap(tenant::getAuthIdTemplate)
+                .map(IdentityTemplate::new)
+                .map(t -> t.apply(getAuthId()))
+                .ifPresent(this::setGeneratedAuthId);
+    }
+
+    /**
+     * Gets the authentication identifier generated by applying the <em>auth-id-template</em> from the tenant's trust
+     * anchor to the client certificate's subject DN.
+     *
+     * @return The generated authentication identifier.
+     */
+    @JsonProperty(value = RegistryManagementConstants.FIELD_GENERATED_AUTH_ID)
+    public final String getGeneratedAuthId() {
+        return generatedAuthId;
```

Comment:
this means that we are exposing this property to external clients, doesn't it?

Reply:
My intention was to include the _generatedAuthId_ for serialization to persist in the database. By setting the `@JsonProperty`, the property is exposed to the external clients and I overlooked it.  I use a different approach now and pushed a commit for it. Would you mind taking a look again.